lib/container_orchestrator.rb

```diff
@@ -58,7 +58,21 @@ def delete_secret(name)
     raise unless e.message =~ /not found/
   end
 
+  def get_pods
+    # TODO: we should set a label for workers we want to monitor so we don't accidentally kill memcached/postgresql pod
+    # :label_selector => "app=manageiq,worker=true"
+    kube_connection.get_pods(pod_options)
+  end
+
+  def watch_pods(resource_version = 0)
+    watcher = kube_connection.watch_pods(pod_options.merge(:resource_version => resource_version))
+    watcher.each { |notice| yield notice }
+  end
+
   private
+  def pod_options
+    @pod_options ||= {:namespace => my_namespace, :label_selector => "app=#{app_name}"}
```

Comment:
Pairing with @Fryguy, we came up with a label we can set for all deployments  `{:"#{app_name}-orchestrated-by" => ENV['POD_NAME']}` in the object definition and we can then do a selector here `app=manageiq,manageiq-orchestrated-by=orchestrator-9f99d8cb9-7mprg` so we'll only get pods that are managed by the orchestrator.  In the orchestrator code, we can then look for all pods that we're managing...